runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -409,6 +425,81 @@ function LanguageTree:_add_injections()
   return query_time
 end
 
+--- @param range boolean|Range?
+--- @return string
+local function range_to_string(range)
+  return type(range) == 'table' and table.concat(range, ',') or tostring(range)
+end
+
+--- @private
+--- @param range boolean|Range?
+--- @param callback fun(err?: string, trees?: table<integer, TSTree>)
+function LanguageTree:_push_async_callback(range, callback)
+  local key = range_to_string(range)
+  self._cb_queues[key] = self._cb_queues[key] or {}
+  local queue = self._cb_queues[key]
+  queue[#queue + 1] = callback
```

Comment:
off-topic: this is the ["event emitter"](https://nodejs.org/en/learn/asynchronous-work/the-nodejs-event-emitter) pattern. Do we need a stdlib util for this?

Reply:
Isn't that basically user autocommands without vim baggage?

Reply:
yeah I think it should probably be a feature of autocmds somehow.